EventFilter/CTPPSRawToDigi/interface/TotemSampicFrame.h

```diff
@@ -0,0 +1,263 @@
+* Authors:
+*   Nicola Minafra
+*
+****************************************************************************/
+
+#ifndef EventFilter_CTPPSRawToDigi_TotemSampicFrame
+#define EventFilter_CTPPSRawToDigi_TotemSampicFrame
+
+#include <vector>
+#include <cstddef>
+#include <cstdint>
+#include <iomanip>
+#include <bitset>
+
+#include "EventFilter/CTPPSRawToDigi/interface/VFATFrame.h" 
+
+enum TotemSampicConstant {
+  NumberOfSamples = 24,
+};
+
+#pragma pack(push,1)
+struct TotemSampicData{
+  uint8_t      samples[TotemSampicConstant::NumberOfSamples];
+   
+  TotemSampicData() {}
+};
+#pragma pack(pop)
+
+#pragma pack(push,1)
+struct TotemSampicInfo{
+  uint8_t       hwId;
+  uint8_t       controlBits[6];
+  uint8_t       FPGATime[5];
+  uint16_t      TimeStampA;
+  uint16_t      TimeStampB;
+  uint16_t      CellInfo;
+  uint8_t       PlaneChannelId;
+  uint8_t       reserved[5];
+   
+  TotemSampicInfo() {}
+};
+#pragma pack(pop)
+
+#pragma pack(push,1)
+struct TotemSampicEventInfo{
+  uint8_t       hwId;
+  uint8_t       L1ATimeStamp[5];
+  uint16_t      bunchNumber;
+  uint32_t      orbitNumber;
+  uint32_t      eventNumber;
+  uint16_t      channelMap;
+  uint16_t      L1ALatency;
+  uint8_t       numberOfSamples;
+  uint8_t       offsetOfSamples;
+  uint8_t       FWVersion;
+  uint8_t       PLLInfo;
+   
+  TotemSampicEventInfo() {}
+};
+#pragma pack(pop)
+
+uint8_t GrayToBinary_8bit(const uint8_t &gcode_data)
+{
+  //b[0] = g[0]   
+  uint8_t binary_byte=(gcode_data & 0x80); //MSB is the same
+
+  //b[i] = g[i] xor b[i-1]
+  binary_byte |= (gcode_data ^ (binary_byte >> 1)) & 0x40;
+  binary_byte |= (gcode_data ^ (binary_byte >> 1)) & 0x20;
+  binary_byte |= (gcode_data ^ (binary_byte >> 1)) & 0x10;
+  binary_byte |= (gcode_data ^ (binary_byte >> 1)) & 0x08;
+  binary_byte |= (gcode_data ^ (binary_byte >> 1)) & 0x04;
+  binary_byte |= (gcode_data ^ (binary_byte >> 1)) & 0x02;
+  binary_byte |= (gcode_data ^ (binary_byte >> 1)) & 0x01;
+
+  return binary_byte;
+}
+
+/** 
+ * This class is intended to handle the timing infromation of SAMPIC in the TOTEM implementation
+**/
+class TotemSampicFrame
+{
+  public:
+    TotemSampicFrame(const uint8_t* chInfoPtr, const uint8_t* chDataPtr, const uint8_t* eventInfoPtr) : 
+    TotemSampicInfo_(nullptr), TotemSampicData_(nullptr), TotemSampicEventInfo_(nullptr), status_(0) {
+      if ( chInfoPtr!=nullptr && chDataPtr!=nullptr && eventInfoPtr!=nullptr) {
+        TotemSampicInfo_ = (TotemSampicInfo*) chInfoPtr;
+        TotemSampicData_ = (TotemSampicData*) chDataPtr;
+        TotemSampicEventInfo_ = (TotemSampicEventInfo*) eventInfoPtr;
+      }
+      if ( TotemSampicEventInfo_->numberOfSamples == TotemSampicConstant::NumberOfSamples || TotemSampicInfo_->controlBits[3] == 0x69 )
+        status_ = 1;
+    }
+    ~TotemSampicFrame() {}
+
+    /// Prints the frame.
+    /// If binary is true, binary format is used.
+    void PrintRaw(bool binary = false) const {
```

Comment:
start local variables and data members with lower case letter. Capitalization is reserved for class names.